scripts/webidl/convert.js

```diff
@@ -134,7 +144,97 @@ const transformer = new Webidl2js({
       };
     }
 
+    if (isSimpleIDLType(idl.idlType, "SVGAnimatedString")) {
+      const SVGAnimatedString = this.addImport("./SVGAnimatedString");
+      checkAttributeNamespace(attrName);
+      const deprecatedAttr = extractAttributeInfo(getExtAttrValue("ReflectDeprecated"));
+      const initialValue = getExtAttrValue("ReflectInitial");
+      if (initialValue !== undefined && typeof initialValue !== "string") {
+        throw new Error("Initial value of SVGAnimatedString must be a string");
+      }
+
+      return {
+        get: `
+          return ${SVGAnimatedString}.create(globalObject, [], {
+            element: ${implObj},
+            attribute: "${attrName}",
+            ${deprecatedAttr !== undefined ?
+              `attributeDeprecatedNamespace: ${JSON.stringify(deprecatedAttr.ns)},
+               attributeDeprecated: ${JSON.stringify(deprecatedAttr.name)},` :
+              ""}
+            ${initialValue !== undefined ?
+              `initialValue: ${JSON.stringify(initialValue)},` :
+              ""}
+          });
+        `
+      };
+    }
+
+    if (isSimpleIDLType(idl.idlType, "SVGAnimatedPreserveAspectRatio")) {
+      const SVGAnimatedPreserveAspectRatio = this.addImport("./SVGAnimatedPreserveAspectRatio");
+      if (attrName !== "preserveAspectRatio") {
+        throw new Error("SVGAnimatedPreserveAspectRatio can only be used with the preserveAspectRatio attribute");
+      }
+
+      return {
+        get: `
+          return ${SVGAnimatedPreserveAspectRatio}.create(globalObject, [], {
+            element: ${implObj}
+          });
+        `
+      };
+    }
+
+    if (isSimpleIDLType(idl.idlType, "SVGAnimatedRect")) {
+      const SVGAnimatedRect = this.addImport("./SVGAnimatedRect");
+      checkAttributeNamespace(attrName);
+
+      return {
+        get: `
+          return ${SVGAnimatedRect}.create(globalObject, [], {
+            element: ${implObj},
+            attribute: "${attrName}"
+          });
+        `
+      };
+    }
+
     throw new Error("Unrecognized reflection type " + idl.idlType.idlType);
+
+    function getExtAttrValue(extAttrName) {
```

Comment:
It's not a big deal, but I think it would be nicer to move this to top-level of the file, and pass in the `idl` input as needed.

Also, the reflection handling is getting pretty big in this file; we might want to consider splitting it up.